.github/workflows/go-test-coverage.yml

```diff
@@ -9,22 +9,39 @@ on:
   pull_request:
     branches: [main, dev, 1.0*, 2.0*, fasttrack/*]
 
-jobs:
+env:
+  EXPECTED_GO_VERSION: "1.20"
 
+jobs:
   build:
     name: Go Test Coverage
     runs-on: ubuntu-latest
     steps:
 
     - name: Set up Go 1.x
-      uses: actions/setup-go@v2
+      uses: actions/setup-go@v4
       with:
-        go-version: 1.19
+        go-version: '${{ env.EXPECTED_GO_VERSION }}'
       id: go
 
+    - name: Check active go version
+      run: |
+        go version && which go
+
     - name: Check out code into the Go module directory
       uses: actions/checkout@v4
 
+    - name: Check go.mod
+      run: |
+        if grep -q "go $EXPECTED_GO_VERSION" ./toolkit/tools/go.mod; then
+          echo "go.mod has correct version ($EXPECTED_GO_VERSION)"
+        else
+          actual_version="$(grep -E '^go [0-9]+\.[0-9]+' ./toolkit/tools/go.mod)"
+          echo "go.mod has bad version expected:$EXPECTED_GO_VERSION, found: $actual_version"
```

Comment:
I'd maybe reverse the message - if someone changed `go.mod`, then it's probably intentional and they just didn't realize they have to update this check as well.